snap/snapcraft.yaml

```diff
@@ -0,0 +1,204 @@
+name: docker
+version: 17.03.1-ce-0
+summary: Docker Linux container runtime
+description: Docker complements kernel namespacing with a high-level API which operates at the process level. It runs unix processes with strong guarantees of isolation and repeatability across servers.
+
+confinement: strict
+grade: stable
+
+plugs:
+  support:
+    interface: docker-support
+  privileged:
+    interface: docker-support
+    privileged-containers: true
+  docker-cli:
+    interface: docker
+slots:
+  docker-daemon:
+    interface: docker
+
+apps:
+  docker:
+    command: docker-wrapper docker
+    plugs:
+      - docker-cli
+      - network
+      - home
+
+  dockerd:
+    command: dockerd-wrapper
+    daemon: simple
+    plugs:
+      - network-bind
+      - firewall-control
+      - support
+      - privileged
+    slots:
+      - docker-daemon
+
+  # TODO determine if we can "provide" a "docker-compose" in PATH directly (since this'll show up as "docker.compose" in the user PATH)
+  compose:
+    command: docker-wrapper docker-compose
+    plugs:
+      - docker-cli
+      - network
+      - home
+  help:
+    command: help
+
+parts:
+  snappy-bins:
+    plugin: shell
+    shell: bash
+    shell-flags: ['-ex']
+    shell-command: |
+      install -d "$SNAPCRAFT_PART_INSTALL/bin"
+      install -t "$SNAPCRAFT_PART_INSTALL/bin" "$SNAPDIR"/bin/*
+    stage-packages:
+      - mount
+
+  go:
+    plugin: gobuild
+    source: https://storage.googleapis.com/golang/go1.7.5.src.tar.gz
+    source-type: tar
+    snap:
+      - -*
+    build-packages:
+      - golang-go
+      - rsync
+      - build-essential
+      - netbase
+      - python3-pyelftools
```

Comment:
Building Go from source seems a little odd -- is this to ensure we're building against the exact version of Go that Docker did for this release?  (I'd put the rationale into a comment so that it's recorded for posterity, which often ends up being me wondering why I did something odd :smile:)